pkg/apis/core/types.go

```diff
@@ -1675,11 +1675,20 @@ type CSIPersistentVolumeSource struct {
 	// ControllerExpandSecretRef is a reference to the secret object containing
 	// sensitive information to pass to the CSI driver to complete the CSI
 	// ControllerExpandVolume call.
-	// This is an alpha field and requires enabling ExpandCSIVolumes feature gate.
+	// This is an beta field and requires enabling ExpandCSIVolumes feature gate.
 	// This field is optional, and may be empty if no secret is required. If the
 	// secret object contains more than one secret, all secrets are passed.
 	// +optional
 	ControllerExpandSecretRef *SecretReference
+
+	// NodeExpandSecretRef is a reference to the secret object containing
+	// sensitive information to pass to the CSI driver to complete the CSI
+	// NodeExpandVolume call.
+	// This is an beta field and requires enabling ExpandCSIVolumes feature gate.
```

Comment:
Heads up for an API reviewer: a new field goes straight to beta. From storage point of view, the field is needed and useful and it's basically a copy of well established `ControllerExpandSecretRef`.

Reply:
We may need to add the field in alpha first. Otherwise a n-1 controller could drop the field on an update.

Reply:
Yes, during an upgrade to "release with this" if someone set the field right away (because it was on by default) then it would get cleared if someone hit the not yet upgraded server.

Is there a justification for "straight to beta" recorded in the KEP discussion or other format?

Reply:
I would prefer not to create the exception here to allow a field to go to beta ungated unless there is a decision by other api reviewers recorded already to do so.  It's unusual enough to give me pause, and michelle is right that it will have unusual implications for upgrades.

Reply:
Thanks all for the review and thoughts. As we are in a conclusion that this has to go through the feature gate starting with Alpha, this has to be tracked for next release.